### PR TITLE
Update node drainTimeout docs

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Update node drainTimeout docs

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml [hunk 1]
```diff
@@ -1174,12 +1174,11 @@ spec:
...
                 description: 'NodeDrainTimeout is the total amount of time that the
-                  controller will spend on draining a node. The default value is 0,
+                  controller will spend on draining a node. The default value is 0s,
                   meaning that the node can be drained without any time limitations.
-                  NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
-                  TODO (alberto): Today changing this field will trigger a recreate
-                  rolling update, which kind of defeats the purpose of the change.
-                  In future we plan to propagate this field in-place. https://github.com/kubernetes-sigs/cluster-api/issues/5880'
+                  NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`.
+                  NOTE: Changes to this field won''t trigger a rollout. It''s applied
```
Comment (jparrill) [inline cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml:1180]: small nit: s/''/'/

File: hack/app-sre/saas_template.yaml [hunk 2]
```diff
@@ -46097,11 +46097,10 @@ objects:
...
                   description: 'NodeDrainTimeout is the total amount of time that
                     the controller will spend on draining a node. The default value
-                    is 0, meaning that the node can be drained without any time limitations.
-                    NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`
-                    TODO (alberto): Today changing this field will trigger a recreate
-                    rolling update, which kind of defeats the purpose of the change.
-                    In future we plan to propagate this field in-place. https://github.com/kubernetes-sigs/cluster-api/issues/5880'
+                    is 0s, meaning that the node can be drained without any time limitations.
+                    NOTE: NodeDrainTimeout is different from `kubectl drain --timeout`.
+                    NOTE: Changes to this field won''t trigger a rollout. It''s applied
```
Comment (jparrill) [inline hack/app-sre/saas_template.yaml:46102]: small nit: s/''/'/

**Review by jparrill — COMMENTED** [review]
/lgtm

File: cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml (hunk 1 above)
Comment (enxebre) [reply cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml:1180]: this is autogenerated, not sure why is generated like that though.

**Review by jparrill — COMMENTED** [review]
/lgtm

**Pull request closed without merging**